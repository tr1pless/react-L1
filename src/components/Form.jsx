import React, { useState} from 'react';
import { Button } from './Button/Button';


export const Form = ({addMessage}) => {
  const [value, setValue] = useState('')
  const [name, setName] = useState('')

  const handleSubmitForm = (e) => {
    e.preventDefault();
    setValue('')
    addMessage(value,name);
    console.log(name)
  }

  return (
    <form onSubmit={handleSubmitForm}>
      <p>Введите Имя</p>
      <input type="text" onChange={e => setName(e.target.value)} value={name} />
      <p>Введите ваше сообщение</p>
      <textarea  
      value={value} 
      onChange={e => setValue(e.target.value)}/>
      <Button onButtonClick={() => {}}/>
    </form>
  );
};
