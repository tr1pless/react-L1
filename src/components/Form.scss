.formImg {
    width: 150px;
    height: 150px;
}

.textfield {
    height: 300px;
    width: 400px;
}
