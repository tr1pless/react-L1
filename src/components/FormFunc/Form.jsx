import React, { useState, useEffect } from 'react';
import { Button } from './Button';
import { Textarea } from './Textarea';
import { Input } from './Input';
import { List } from './List';

export const Form = () => {
  const [defaultMessage] = useState('Ожидайте ответа');
  const [bot] = useState('Автоответ');
  const [message, setMessage] = useState('');
  const [author, setAuthor] = useState('');
  const [messageList, setMessageList] = useState([]);
  const handleClick = () => {
    setMessageList([...messageList, { message, author, botMessage: false }]);
    setMessage('');
    setAuthor('');
  };
  const changeАuthor = (event) => {
    setAuthor(event.target.value);
  };
  const changeMessage = (event) => {
    setMessage(event.target.value);
  };
  useEffect(() => {
    if (messageList.length && !messageList[messageList.length - 1].botMessage) {
      setTimeout(() => {
        setMessageList([
          ...messageList,
          {
            message: defaultMessage,
            author: bot,
            botMessage: true,
          },
        ]);
      }, 1000);
    }
  }, [messageList.length]);

  return (
    <div className="content">
      <List messages={messageList} />
      <div className="form">
        <Input
          placeholder="Введите ваше имя"
          author={author}
          change={changeАuthor}
        />
        <Textarea change={changeMessage} message={message} />
        <Button name="Отправить" click={handleClick} />
      </div>
    </div>
  );
};
