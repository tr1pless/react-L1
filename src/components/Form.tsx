import React, { useState, memo } from "react";
import { Button } from "./Button/Button";
import Input from "@mui/material/TextField";
import TextField from "@mui/material/TextField";
import { ChatList } from "./Chats/ChatList";
import "./Form.scss";

interface FormProps {
  addMessage: (value: string, name: string) => void;
}

export const Form = memo<FormProps>(({ addMessage }) => {
  const [value, setValue] = useState("");
  const [name, setName] = useState("");

  const handleSubmitForm = (event: React.FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    setValue("");
    addMessage(value, name);
  };

  return (
    <form onSubmit={handleSubmitForm}>
      <Input
        label="Введите имя"
        variant="standard"
        onChange={(event) => setName(event.target.value)}
        value={name}
      />
      <TextField
        id="outlined-multiline-static"
        rows={4}
        multiline
        placeholder="Введите текст сообщения"
        className="textfield"
        autoFocus={true}
        value={value}
        onChange={(event) => setValue(event.target.value)}
      />
      <Button onButtonClick={() => {}} />
    </form>
  );
});
