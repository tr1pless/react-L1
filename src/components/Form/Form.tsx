import React, { useState, FC, memo } from "react";
import { Input } from "@mui/material";
import { Button } from "./components/Button";
import { useDispatch } from "react-redux";
import { addMessageWithReply } from "../../store/chats/actions";
import { useParams } from "react-router-dom";
import { AUTHOR } from "../../constants";

export const Form: FC = memo(() => {
  const [value, setValue] = useState("");
  const dispatch = useDispatch();
  const { chatId } = useParams();

  const handleSubmitForm = (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    if (chatId && value) {
      dispatch(
        addMessageWithReply(chatId, { text: value, author: AUTHOR.USER })
      );
    }
    setValue("");
  };

  return (
    <form onSubmit={handleSubmitForm}>
      <Input
        type="text"
        value={value}
        onChange={(e) => setValue(e.target.value)}
      />
      <Button disabled={!value} />
    </form>
  );
});
