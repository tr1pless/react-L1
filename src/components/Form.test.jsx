import React from 'react'
import { Form } from './Form'
import {render} from '@testing-library/react'
import '@testing-library/jest-dom'


describe('render with snapshot', (props) => {
    it('need to render form with snapshot', () => {
      const { asFragment } = render(<Form {...props} />)
      expect(asFragment()).toMatchSnapshot()
    })
  }) 