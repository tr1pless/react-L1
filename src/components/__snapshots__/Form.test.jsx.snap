// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render with snapshot need to render form with snapshot 1`] = `
<DocumentFragment>
  <form>
    <p>
      Введите Имя
    </p>
    <input
      type="text"
      value=""
    />
    <p>
      Введите ваше сообщение
    </p>
    <textarea />
    <button
      style="background-color: red;"
      type="submit"
    >
      click
    </button>
  </form>
</DocumentFragment>
`;
